File: MantenimientoDaxDirectorio/MantenimientoDirectorio/LeerEmpleado.vb
Imports DattCom
Friend Class LeerEmpleado
    Friend Shared Function leerEmpleado(Form As DEEP01, QueCodigo As String, TipCodSyscod() As String, LosCambios As ClassCambios, codigoDirectorio() As String) As Boolean
        Dim Buscod As New Syscod.ManSysnetClass
        Dim Busnom As New EmpNomR.AdcNomb()
        Dim classEmpleado As New dbEmpleado(datosEmpresa.strConxAdcom)
        classEmpleado = dbEmpleado.Buscar(" codigoempleado = '" & QueCodigo & "' ")
        If IsDBNull(classEmpleado) Then Return False
        If IsNothing(classEmpleado) Then Return False

        With classEmpleado
            Form.cmbCentroCostoRol.Text = .centroCosto
            LosCambios.AntCentroCosto = .centroCosto

            Form.cmbSeccionRol.Text = .Libretamilitar
            LosCambios.AntSeccion = .Libretamilitar

            Form.CodSuc = .Sucursalrol
            LosCambios.AntSucursal = Form.CodSuc
            Form.cmbSucursalRol.SelectedValue = .Sucursalrol

            Form.cmbDepartamentoRol.SelectedValue = .Departamento
            LosCambios.AntDepartamento = .Departamento

            Form.cmbCargoRol.SelectedValue = .Cargorol
            LosCambios.AntCargo = .Cargorol

            Form.fingreso.Value = .FechaIngreso
            LosCambios.AntFechaIngreso = .FechaIngreso

            Form.fsalida.Value = .FechaSalida
            LosCambios.AntFechaSalida = .FechaSalida

            Form.fJubilacion.Value = .FechaJubilacion

            Form.nivelrol.Text = .NivelRol.ToString()
            Form.Cuenta2.Text = .CtbRol_0_
            Form.Cuenta3.Text = .CtbRol_1_

            Dim tp As String = .TipoPago
            If tp = "P" Then
                Form.rolproduccion.Checked = True
            ElseIf tp = "J" Then
                Form.roldiario.Checked = True
            ElseIf tp = "S" Then
                Form.rolmensual.Checked = True
            ElseIf tp = "H" Then
                Form.RolHoras.Checked = True
            End If

            If .AcreditarBanco Then Form.AcreditaBanco.CheckState = Windows.Forms.CheckState.Checked Else Form.AcreditaBanco.CheckState = Windows.Forms.CheckState.Unchecked
            codigoDirectorio(2) = .BancoRol : Form.txtNomBanco.Text = Busnom.RetornaNombreDirectorio(codigoDirectorio(2), datosEmpresa.strConxAdcom)

            Form.Lcod11.Text = Buscod.QueNombre(.TipoSalario, TipCodSyscod(11))
            Form.txtHorasJornadaSemanal.Text = .HorSemanaParcial.ToString()
            Form.valorsueldo.Text = .ValorSueldo.ToString()
            LosCambios.AntSueldo = .ValorSueldo
            Form.txtHorasJornadaSemanal.Text = CStr(.HorSemanaParcial)
            Form.Nroctabancorol.Text = CStr(.NroCtaBancoRol)
            If .TipoCuentaBanco = "A" Then
                Form.ctaahorrosrol.Checked = True
            Else
                Form.ctaahorrosrol.Checked = False
            End If
            If .TipoCuentaBanco = "C" Then
                Form.ctacorrienterol.Checked = True
            Else
                Form.ctacorrienterol.Checked = False
            End If

            Form.Grupo1.Text = classEmpleado.Grupo_1
            Form.Grupo2.Text = classEmpleado.Grupo_2
            Form.Grupo3.Text = classEmpleado.Grupo_3
            Form.Grupo4.Text = classEmpleado.Grupo_4
            Form.Grupo5.Text = classEmpleado.Grupo_5
            Form.Grupo6.Text = classEmpleado.Grupo_6

            Select Case .EstadoRol
                Case "A"
                    Form.activorol.Checked = True
                Case "C"
                    Form.Cesanterol.Checked = True
                Case "E"
                    Form.Eliminadorol.Checked = True
                Case "J"
                    Form.Jubilado.Checked = True
            End Select

            Form.cmbModuloRol.Text = .Idusuario
            Form.CodBimetrico.Text = .codigoBiometrico
        End With

        Return True
    End Function
    Public Shared Function ExisteCodigo(codigo As String) As Boolean
        Dim resp As Boolean = False
        Using conn As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
            conn.Open()
            Dim comm As New SqlClient.SqlCommand("select codigo from identificacion where codigo = '" + codigo + "'", conn)
            Dim dr As SqlClient.SqlDataReader = comm.ExecuteReader()
            If dr.Read Then resp = True
            Return resp
            dr.Close()
            comm.Dispose()
            conn.Close()
        End Using

    End Function
    Public Shared Function ExisteCedula(codigo As String, cedula As String) As Boolean
        Dim resp As Boolean = False
        Using conn As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
            conn.Open()
            Dim comm As New SqlClient.SqlCommand("select codigo from identificacion where cedulaidentidadruc='" + cedula & "' and codigo <> '" & codigo & "' ", conn)
            Dim dr As SqlClient.SqlDataReader = comm.ExecuteReader()
            If dr.Read Then resp = True
            Return resp
            dr.Close()
            comm.Dispose()
            conn.Close()
        End Using
    End Function


End Class

File: MantenimientoDaxDirectorio/MantenimientoDirectorio/LeerEmpleado.cs
using System;
using DattCom;

namespace directMnt
{
	internal class LeerEmpleado
	{
		internal static bool leerEmpleado(DEEP01 Form, string QueCodigo, string[] TipCodSyscod, ClassCambios LosCambios, string[] codigoDirectorio)
		{
			var Buscod = new Syscod.ManSysnetClass();
			var Busnom = new EmpNomR.AdcNomb();
			var classEmpleado = new dbEmpleado(datosEmpresa.strConxAdcom);
			classEmpleado = dbEmpleado.Buscar(" codigoempleado = '" + QueCodigo + "' ");
			if (classEmpleado is DBNull)
				return false;
			if (classEmpleado == null)
				return false;

			Form.cmbCentroCostoRol.Text = classEmpleado.centroCosto;
			LosCambios.AntCentroCosto = classEmpleado.centroCosto;

			Form.cmbSeccionRol.Text = classEmpleado.Libretamilitar;
			LosCambios.AntSeccion = classEmpleado.Libretamilitar;

			Form.CodSuc = classEmpleado.Sucursalrol;
			LosCambios.AntSucursal = Form.CodSuc;
			Form.cmbSucursalRol.SelectedValue = classEmpleado.Sucursalrol;

			Form.cmbDepartamentoRol.SelectedValue = classEmpleado.Departamento;
			LosCambios.AntDepartamento = classEmpleado.Departamento;

			Form.cmbCargoRol.SelectedValue = classEmpleado.Cargorol;
			LosCambios.AntCargo = classEmpleado.Cargorol;

			Form.fingreso.Value = classEmpleado.FechaIngreso;
			LosCambios.AntFechaIngreso = classEmpleado.FechaIngreso;

			Form.fsalida.Value = classEmpleado.FechaSalida;
			LosCambios.AntFechaSalida = classEmpleado.FechaSalida;

			Form.fJubilacion.Value = classEmpleado.FechaJubilacion;

			Form.nivelrol.Text = classEmpleado.NivelRol.ToString();
			Form.Cuenta2.Text = classEmpleado.CtbRol_0_;
			Form.Cuenta3.Text = classEmpleado.CtbRol_1_;

			string tp = classEmpleado.TipoPago;
			if (tp == "P")
			{
				Form.rolproduccion.Checked = true;
			}
			else if (tp == "J")
			{
				Form.roldiario.Checked = true;
			}
			else if (tp == "S")
			{
				Form.rolmensual.Checked = true;
			}
			else if (tp == "H")
			{
				Form.RolHoras.Checked = true;
			}

			if (classEmpleado.AcreditarBanco)
				Form.AcreditaBanco.CheckState = System.Windows.Forms.CheckState.Checked;
			else
				Form.AcreditaBanco.CheckState = System.Windows.Forms.CheckState.Unchecked;
			codigoDirectorio[2] = classEmpleado.BancoRol;
			Form.txtNomBanco.Text = Busnom.RetornaNombreDirectorio(codigoDirectorio[2], datosEmpresa.strConxAdcom);

			Form.Lcod11.Text = Buscod.QueNombre(classEmpleado.TipoSalario, TipCodSyscod[11]);
			Form.txtHorasJornadaSemanal.Text = classEmpleado.HorSemanaParcial.ToString();
			Form.valorsueldo.Text = classEmpleado.ValorSueldo.ToString();
			LosCambios.AntSueldo = (double)classEmpleado.ValorSueldo;
			Form.txtHorasJornadaSemanal.Text = classEmpleado.HorSemanaParcial.ToString();
			Form.Nroctabancorol.Text = classEmpleado.NroCtaBancoRol;
			if (classEmpleado.TipoCuentaBanco == "A")
			{
				Form.ctaahorrosrol.Checked = true;
			}
			else
			{
				Form.ctaahorrosrol.Checked = false;
			}
			if (classEmpleado.TipoCuentaBanco == "C")
			{
				Form.ctacorrienterol.Checked = true;
			}
			else
			{
				Form.ctacorrienterol.Checked = false;
			}

			Form.Grupo1.Text = classEmpleado.Grupo_1;
			Form.Grupo2.Text = classEmpleado.Grupo_2;
			Form.Grupo3.Text = classEmpleado.Grupo_3;
			Form.Grupo4.Text = classEmpleado.Grupo_4;
			Form.Grupo5.Text = classEmpleado.Grupo_5;
			Form.Grupo6.Text = classEmpleado.Grupo_6;

			switch (classEmpleado.EstadoRol ?? "")
			{
				case "A":
					{
						Form.activorol.Checked = true;
						break;
					}
				case "C":
					{
						Form.Cesanterol.Checked = true;
						break;
					}
				case "E":
					{
						Form.Eliminadorol.Checked = true;
						break;
					}
				case "J":
					{
						Form.Jubilado.Checked = true;
						break;
					}
			}

			Form.cmbModuloRol.Text = classEmpleado.Idusuario;
			Form.CodBimetrico.Text = classEmpleado.codigoBiometrico;

			return true;
		}
		public static bool ExisteCodigo(string codigo)
		{
			bool resp = false;
			using (var conn = new System.Data.SqlClient.SqlConnection(datosEmpresa.strConxAdcom))
			{
				conn.Open();
				var comm = new System.Data.SqlClient.SqlCommand("select codigo from identificacion where codigo = '" + codigo + "'", conn);
				var dr = comm.ExecuteReader();
				if (dr.Read())
					resp = true;
				return resp;
				dr.Close();
				comm.Dispose();
				conn.Close();
			}

		}
		public static bool ExisteCedula(string codigo, string cedula)
		{
			bool resp = false;
			using (var conn = new System.Data.SqlClient.SqlConnection(datosEmpresa.strConxAdcom))
			{
				conn.Open();
				var comm = new System.Data.SqlClient.SqlCommand("select codigo from identificacion where cedulaidentidadruc='" + cedula + "' and codigo <> '" + codigo + "' ", conn);
				var dr = comm.ExecuteReader();
				if (dr.Read())
					resp = true;
				return resp;
				dr.Close();
				comm.Dispose();
				conn.Close();
			}
		}


	}
}